met-web/src/components/metadataManagement/TaxonEditForm.tsx

```diff
@@ -0,0 +1,325 @@
+            is: (value: string) => TaxonTypes[value as keyof typeof TaxonTypes].supportsPresetValues,
+            then: yup.mixed().when('freeform', {
+                is: false,
+                then: yup
+                    .array()
+                    .of(taxonType.yupValidator ?? yup.mixed())
+                    .required('Preset value is required'),
+                otherwise: yup.array().of(taxonType.yupValidator ?? yup.mixed()),
+            }),
+            otherwise: yup.mixed().strip(),
+        }),
+    });
+
+    const onSubmit: SubmitHandler<MetadataTaxon> = async (data, event) => {
+        let formErrors: { [key: string]: string } = {};
+        // These fields don't always apply to all taxon types, so
+        // if they are unsupported, we set them to the defaults
+        if (!taxonType.supportsFreeform) data.freeform = false;
+        if (!taxonType.supportsMulti) data.one_per_engagement = true;
+        if (!taxonType.supportsPresetValues) data.preset_values = [];
+        try {
+            await schema.validate(data, { abortEarly: false });
+            // Catch clause variable type annotation must be 'any' or 'unknown' if specified
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: yup.ValidationError | any) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            formErrors = error.inner.reduce((errors: { [key: string]: string }, innerError: any) => {
+                errors[innerError.path] = innerError.message;
+                return errors;
+            }, {});
+            // Set errors for each field in formState
+            Object.keys(formErrors).forEach((fieldName) => {
+                setError(fieldName as keyof MetadataTaxon, {
+                    type: 'validate',
+                    message: formErrors[fieldName],
+                });
+            });
+        }
+
+        if (Object.keys(formErrors).length) {
+            console.log('Form errors:', formErrors);
+            dispatch(
+                openNotification({ text: 'Please correct the highlighted errors before saving.', severity: 'error' }),
+            );
+            return false;
+        }
+
+        const updatedTaxon: MetadataTaxon = {
+            ...taxon,
+            name: data.name,
+            description: data.description,
+            freeform: data.freeform || data.preset_values?.length === 0,
+            one_per_engagement: data.one_per_engagement || !TaxonTypes[data.data_type ?? 'text'].supportsMulti,
+            data_type: data.data_type,
+            preset_values: TaxonTypes[data.data_type ?? 'text'].supportsPresetValues ? data.preset_values : [],
+        };
+
+        updateMetadataTaxon(updatedTaxon);
+    };
+
+    const handleKeys = (event: React.KeyboardEvent) => {
+        // Handle as many key combinations as possible
+        if ((event.ctrlKey || event.metaKey || event.altKey) && event.key === 'Enter') {
```

Comment:
This is a really nice touch. I wasn't aware that keyboard-only users expected to be able to use CTRL + Enter to submit a form, for example